### DISCUSSION TITLE
Getting errors when trying to run my index.js file on a remote linux server.

### DISCUSSION THREAD

**Discussion by AnimeshDeb** [withcatai/node-llama-cpp#189] [Q&A] [answered]
I have a index.js file utilizing a llama 2 chat model and it works fine when I run it on my localhost. However, when I try to run my file on a remote server that I am connected to, the code does not run due to errors. I have attached images of the errors that I am getting. I also did npm install node-llama-cpp in the directory that my index.js file and my ai model is located in with respect to the remote server.

<img width="1243" alt="ErrorImage1" src="https://github.com/withcatai/node-llama-cpp/assets/102705914/424c4526-60ea-49ec-bf86-fe9906ef3b77">
<img width="1243" alt="ErrorImage2" src="https://github.com/withcatai/node-llama-cpp/assets/102705914/f96f4ab9-7145-40a9-8f2e-7325fc63d2ba">
<img width="1243" alt="ErrorImage3" src="https://github.com/withcatai/node-llama-cpp/assets/102705914/9618277e-94e8-4eaa-82db-88a07430cd5a">
<img width="1264" alt="ErrorImage4" src="https://github.com/withcatai/node-llama-cpp/assets/102705914/aa53c3a2-ccb5-43d2-804e-b1864eaacca3">
<img width="1269" alt="ErrorImage5" src="https://github.com/withcatai/node-llama-cpp/assets/102705914/3c3a176d-264d-4aac-8ed0-ea5819a584f7">
<img width="1265" alt="ErrorImage6" src="https://github.com/withcatai/node-llama-cpp/assets/102705914/8238d442-05f4-42e8-be6f-a669094392ed">

**Comment by giladgd** [comment, marked as answer]
@AnimeshDeb It seems prebuilt binary fails to load on your Linux machine because the `glibc` version is older than `2.29`, and that some C/C++ libraries are missing.

`glibc 2.29` was released in 2019, so your Linux distro may be too outdated to run the prebuilt binaries; updating your distro's latest LTS version may help.
To make the build succeed you need some missing libraries; ChatGPT suggests that you run `sudo apt install build-essential` to install the missing `stdatomic.h` file, but it may not suffice; updating to the latest LTS version of your distro is most likely the right solution.

Also, try to use the latest version of the [version 3 beta](https://github.com/withcatai/node-llama-cpp/pull/105) as it detects various incompatibilities in the current system and explains the issues with potential solutions when applicable.
Always attach textual logs instead of screenshots, as screenshots make it harder to investigate.
